spec/models/gutentag/tag_spec.rb

```diff
@@ -63,5 +63,36 @@
       tag = Gutentag::Tag.create(:name => "Pancakes")
       expect(tag.errors[:name].length).to eq(1)
     end
+
+    context "length of name" do
+      it "validates the length of the name" do
+        expect_any_instance_of(ActiveRecord::ConnectionAdapters::Column).
```

Comment:
I think what I'd prefer here is to instead make these specs avoid any stubbing, and have each of these two specs run conditionally, depending on the column situation. Travis runs the suite across MySQL, PostgreSQL and SQLite, so I'd expect each of the two situations to be covered.

Or maybe, we run this first spec against MySQL only, and the other if ENV["DATABASE"] is _not_ `mysql`?

I'd prefer either of these approaches, as that removes the need for the spec to understand the internals of Rails.

Reply:
I thought about that as well, as this also would redeem the need of reloading the class and everything, however my reasoning to go with stubbing was that for whatever reason, the limit of the field can be changed (the user can write a migration that will change the limit for example) and this tests makes it explicit where the limit is coming from.

But since this only happens for MySQL by default, we can do it the way you proposed.

Or, something that would get rid of stubbing, is that we check the schema during the test and run the one test or the other, something like this:

```ruby
if Gutentag::Tag.columns_hash['name'].limit.present?
  # run test that checks that the validation is in place
else
  # run test that checks that no validation is in place
end
```

which has the downside of repeating the implementation a bit, but makes it more explicit.

What do you prefer?

Reply:
I share your concern about your suggestion repeating the implementation, so I'm leaning more towards using `ENV["DATABASE"] == "mysql"`.

Do _all_ string columns in MySQL have an enforced length? i.e. can we rely on there always being a limit if we're running the tests against MySQL? Or do we need to interrogate it for certain options/flags being set?

Reply:
It seems to be the default when you do t.string in a migration without setting a limit, see my comment here: https://github.com/pat/gutentag/issues/40#issuecomment-351402575

So let's go with ENV['DATABASE'] and conditional tests I'd say.

Do you happen to know if GitHub still loses comments when I force push?

Reply:
I would expect the comments to stick around (though hidden if the code they're pointing to no longer exists) - but I don't force push often enough to be sure. 🤷‍♂️ 

I was wondering if MySQL insists on a length (rather than it just being a default) - and it seems this is indeed the case. From my quick glance through [the MySQL docs](https://dev.mysql.com/doc/refman/5.7/en/create-table.html), a `VARCHAR` column must always have a specified length. So, I therefore think it's safe to have a test that uses the default length running when we're using MySQL, and a test for no length for other databases.